### PR TITLE
parse bpf program type

### PR DESCRIPTION
**Pull request by roikol**
add helper to parse bpf program type

### CODE REVIEW

**Comment by rafaeldtinoco** [comment]
@geyslan (To check the unit tests error regarding use of inexistent OS dynamic library, instead of using the static lib we build).

File: helpers/argumentParsers.go [hunk 1]
```diff
@@ -1705,3 +1707,7 @@ func ParseGetSocketOption(rawValue uint64) (SocketOptionArgument, error) {
 	}
 	return v, nil
 }
+
+func ParseBpfProgType(rawValue uint64) (libbpfgo.BPFProgType, error) {
+	return libbpfgo.BPFProgType(rawValue), nil
```
Comment (grantseltzer) [inline helpers/argumentParsers.go:1712]: We should have a check here to make sure that this is a valid BPF program type. For example, `ParseSocketDomainArgument` has a map of known types. We could implement something similar in libbpfgo.go. What do you think?

**Review by grantseltzer — CHANGES_REQUESTED** [review]

File: helpers/argumentParsers.go (hunk 1 above)
Comment (roikol) [reply helpers/argumentParsers.go:1712]: i agree. it's more correct to have this check here.
in the `String()` function of `libbpfgo.BPFProgType` we have something like that, but the helper parsing function should return an error if the value is not a valid BPF program type.

i'll push an update soon

File: libbpfgo.go [hunk 2]
```diff
@@ -1201,6 +1203,41 @@ func (b BPFProgType) String() (str string) {
 	return str
 }
 
+var BpfProgTypeMap = map[uint64]BPFProgType{
```
Comment (grantseltzer) [inline libbpfgo.go:1206]: I think this map should live in `helpers` next to the ParseBPFProgType() function, this way this map doesn't have to be exported.

File: helpers/argumentParsers_test.go [hunk 3]
```diff
@@ -147,3 +148,37 @@ func TestParseGetSocketOption(t *testing.T) {
 		})
 	}
 }
+
+func TestParseBpfProgType(t *testing.T) {
```
Comment (grantseltzer) [inline helpers/argumentParsers_test.go:152]:
```suggestion
func TestParseBPFProgType(t *testing.T) {
```

File: helpers/argumentParsers.go [hunk 4]
```diff
@@ -1705,3 +1707,11 @@ func ParseGetSocketOption(rawValue uint64) (SocketOptionArgument, error) {
 	}
 	return v, nil
 }
+
+func ParseBpfProgType(rawValue uint64) (libbpfgo.BPFProgType, error) {
```
Comment (grantseltzer) [inline helpers/argumentParsers.go:1711]:
```suggestion
func ParseBPFProgType(rawValue uint64) (libbpfgo.BPFProgType, error) {
```

**Comment by geyslan** [comment]
> @geyslan (To check the unit tests error regarding use of inexistent OS dynamic library, instead of using the static lib we build).

https://github.com/aquasecurity/libbpfgo/pull/212 merged.

**Review by grantseltzer — APPROVED** [review]
LGTM!

**Pull request merged**